### DISCUSSION TITLE
Excluding xaml files in build based on target

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#15227] [General]
Do you have any suggestions on how to exclude xaml files from a build based on the target?  Even if you need to exclude a directory would be acceptable.

thanks

**Comment by jeromelaban** [comment]
Great question. Excluding files is done though using multiple exclusion groups, as such:

```xml
<ItemGroup>
	<Page Remove="$(MSBuildThisFileDirectory)MyFolder\**" />
	<Compile Remove="$(MSBuildThisFileDirectory)MyFolder\**" />
	<None Include="$(MSBuildThisFileDirectory)MyFolder\**" />
</ItemGroup>
```

You can use a `Condition` on the `ItemGroup` like `$(TargetFramework.Contains('-android'))` to remove the files.

**Reply by cconner100** [reply]
Thanks,
What about these statements i see in the csproj

  <ItemGroup>
    <AdditionalFiles Include="TestViews\ContactPage.xaml">
      <XamlRuntime>$(DefaultXamlRuntime)</XamlRuntime>
      <SourceItemGroup>Page</SourceItemGroup>
      <Generator>MSBuild:Compile</Generator>
    </AdditionalFiles>
<ItemGroup>

do i need to do something with these or the above Page directive will work

**Reply by cconner100** [reply]
Also i tried your above suggestion in my project, i get the following errors
 when i add this to my project

```
  <ItemGroup Condition="$(TargetFramework.Contains('-windows')) != true">
    <Page Remove="**\Windows\**" />
    <Compile Remove="**\Windows\**" />
    <None Include="**\Windows\**" />
  </ItemGroup>
```

then the error is

E:\ZiiDMS\ZiiDMSApp\Client\ZiiDMSApp\ZiiDMSApp.Common\obj\Debug\net8.0-windows10.0.19041\Views\Windows\NavigationContainer.xbf" because it was not found.

**Comment by cconner100** [comment]
Here is a small repro on what i have tried and its not working. If someone could give me a solution that would be great,

There are 2 directories under presentation Windows and MacOS, I only want the windows directory to compile when compiling for windows and exclude the MacOS.  

thanks
[excluderepo.zip](https://github.com/unoplatform/uno/files/14087885/excluderepo.zip)

**Reply by jeromelaban** [reply]
The exclusions must be placed at the end of the file, in order to [cover the other groups](https://github.com/unoplatform/uno.templates/blob/release/stable/5.0/src/Uno.Templates/content/unoapp/MyExtensionsApp.1/MyExtensionsApp.1.csproj#L285-L292) that are including files.

A few other modifications are needed to make everything compile (missing event handlers and MainPage being Catalyst+Windows specific), otherwise the sample compiles.

**Reply by cconner100** [reply]
Yep that fixed it thanks
